circle.yml

```diff
@@ -19,3 +23,4 @@ test:
       --with-xunit --xunit-file=$CIRCLE_TEST_REPORTS/nosetests/tests.xml
       --with-coverage --cover-min-percentage=37 --cover-html --cover-erase --cover-package=history
       --cover-html-dir=$CIRCLE_TEST_REPORTS/coverage/history
+    - tox
```

Comment:
This is wrong as Tox is a wrapper to run commands on multiple python versions.
Tox must run the `nosetests` command with py2.7 and py3.6: http://tox.readthedocs.io/en/latest/example/nose.html.

Your current tox does nothing. The current `nosetests` command runs the test for only 1 python version

Reply:
You have to move the nosetest command into the tox file: http://tox.readthedocs.io/en/latest/example/pytest.html

Reply:
@xav-b has a PR in kp-scrapers to move it to Py3. You can refer it here - https://github.com/Kpler/kp-scrapers/pull/174/files#diff-b91f3d5bd63fcd17221b267e851608e8R22

And to see what the `make test` command is doing, you can refer here - https://github.com/Kpler/kp-scrapers/blob/master/tools/devops/Python.mk

This will probably help you to understand and implement what @charles-vdulac said properly.

Reply:
That's what I did @charles-vdulac , please have a look at the CircleCi build again.

Reply:
You should move the full command: 

```
nosetests -v --with-timer
       --with-xunit --xunit-file=$CIRCLE_TEST_REPORTS/<python-version>/nosetests/tests.xml
       --with-coverage --cover-min-percentage=37 --cover-html --cover-erase --cover-package=history
       --cover-html-dir=$CIRCLE_TEST_REPORTS/<python-version>/coverage/history
```